frostsnap_coordinator/src/usb_serial_manager.rs

```diff
@@ -536,6 +554,92 @@ impl UsbSerialManager {
     pub fn devices_by_ports(&self) -> &HashMap<String, Vec<DeviceId>> {
         &self.reverse_device_ports
     }
+
+    /// The firmware digest the device has declared it has
+    pub fn firmware_digest_for_device(&self, device_id: DeviceId) -> Option<FirmwareDigest> {
+        self.device_ports
+            .get(&device_id)
+            .map(|device_port| device_port.firmware_digest)
+    }
+
+    pub fn upgrade_bin(&self) -> FirmwareBin {
+        self.firmware_bin
+    }
+
+    pub fn run_firmware_upgrade(&mut self) -> impl Iterator<Item = anyhow::Result<f32>> + '_ {
+        let n_chunks = ceil_div(
+            self.firmware_bin.bin.len() as u32,
+            FIRMWARE_UPGRADE_CHUNK_LEN,
```

Comment:
On the device we're checking the partition size is divisible by this chunk length.
If i recall correctly, our firmware size is going to be equal to the partition size.
So would this binary also be exactly divisible by this chunk length, rather than ceil_div?

Reply:
Yes. We don't pad it to partition length on the wire.